pulser/tests/test_sequence.py

```diff
@@ -81,6 +81,39 @@ def test_channel_declaration():
         seq2.declare_channel('ch3', 'rydberg_global')
 
 
+def test_magnetic_field():
+    seq = Sequence(reg, MockDevice)
+    with pytest.raises(AttributeError, match="only defined when the sequence "
+                                             "is in 'XY Mode'."):
+        seq.magnetic_field
+    seq.declare_channel('ch0', 'mw_global')    # seq in XY mode
+    assert seq.magnetic_field == (0., 0., 1.)    # mag field is the default
+    seq.set_magnetic_field(bx=1., by=-1., bz=0.5)
+    assert seq.magnetic_field == (1., -1., 0.5)
+    assert seq._empty_sequence
+    seq.add(Pulse.ConstantPulse(100, 1, 1, 0), 'ch0')
+    assert not seq._empty_sequence
+    with pytest.raises(ValueError, match="can only be set on an empty seq"):
+        seq.set_magnetic_field(0., 0., 0.)
```

Comment:
Null vectors should raise an error, otherwise there would be no splitting of the Rydberg levels.

Reply:
As long as the there is a non-zero component, it's fine?